### ISSUE TITLE
YAMLGenerator: multiline string will always be quoted when line has trailing space

### ISSUE THREAD

**Issue by njank** [FasterXML/jackson-dataformats-text#366]
I am generating a yml file with the below settings:
```
ObjectMapper mapper = new ObjectMapper(new YAMLFactory()
	.enable(YAMLGenerator.Feature.LITERAL_BLOCK_STYLE)
	.enable(YAMLGenerator.Feature.MINIMIZE_QUOTES)
);
```

This will write multiline strings as following:
```
    - replaceInFiles:
        replacement: test
                     123
```

However if i put a space between "test" and the following line break it will quote the string as if MINIMIZE_QUOTES was disabled:
```
    - replaceInFiles:
        replacement: "test \n\n123\n"
```

**Comment by nylend95** [comment]
I have the same issue.
Example YAML:
```yaml
someKey:
  otherKey.properties: |-
    # comment with trailing whitespace  
    property1=my-value
```
This will be converted to 
```yaml
someKey:
  otherKey.properties: "# comment with trailing whitespace \nproperty1=my-value"
```

See example Maven project: 
[jackson-yaml.zip](https://github.com/user-attachments/files/16964015/jackson-yaml.zip)

**Build and run**
1. Download zip
2. Unzip
3. Run `mvn clean package`
4. Run `java -jar target/jackson-yaml-issue-1-jar-with-dependencies.jar`

**Comment by cowtowncoder** [comment]
@nylend95 And this is with 2.17.2?

**Comment by nylend95** [comment]
Yes! Forgot to include that in the description, but as you can see in my example, I have configured `2.17.2` in the `pom.xml`.

**Comment by cowtowncoder** [comment]
On issue itself, just to make sure I understand what is expected: it is to keep "use literal block style" usage, even if "minimize quotes" is desired?

```
.enable(YAMLGenerator.Feature.LITERAL_BLOCK_STYLE)
	.enable(YAMLGenerator.Feature.MINIMIZE_QUOTES)
```

@nylend95 I don't see the first setting in your example tho?

I am not sure if above is doable, might be? (or whether it's easy to do if so).
I don't have much time to focus on this either, so contributions (PR) would be the best way to move forward (I do take time to review contributions).

But one thing that is often helpful is to contribute PR for failing test (under `yaml/src/test/java/...../failing/`) even without fix: having easily runnable reproduction is often surprisingly helpful wrt actual fixing. Plus ensures fix actually works.

**Comment by nylend95** [comment]
Ah, yes! Our use cases is to keep block style where suitable, and no quotes on single lines if possible. I have yet to dig into the code to see how/if these settings are triggering different behavior. 

With both of these features enabled, I still see the same behavior as before. 
```java
JsonMapper.builder(new YAMLFactory()
                .enable(YAMLGenerator.Feature.LITERAL_BLOCK_STYLE)
                .enable(YAMLGenerator.Feature.MINIMIZE_QUOTES)
                .enable(YAMLGenerator.Feature.INDENT_ARRAYS_WITH_INDICATOR)
                .enable(YAMLGenerator.Feature.ALWAYS_QUOTE_NUMBERS_AS_STRINGS)
                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER))
                .enable(JsonReadFeature.ALLOW_SINGLE_QUOTES)
                .build().findAndRegisterModules();
```

The block style string is converted to a single line when at least one of the lines end with a whitespace. I don't know if it should be considered a bug or not, but I find it a bit weird behavior that a change in a trailing space on a single line should trigger that the field will be converted to a single line.

If you/FasterXML decides that this behavior is wrong, I can get a PR up for the failing test and then we can take it from there.

**Comment by cowtowncoder** [comment]
From description this does appear to be a flaw. One thing to note is that the low-level encoding is by SnakeYAML so it may be down to question of interaction between Jackson YAML module and SnakeYAML generator.
But hopefully it's just a simple logic flaw (missing handling of some combination of settings, most likely) that can be corrected.

And yes, we could use help here with failing test!

**Comment by cowtowncoder** [comment]
Yes, if it is something reproducible with SnakeYAML. If it turns out it is due to way Jackson uses SnakeYAML then needs to be fixed here of course.

**Comment by njank** [comment]
> But one thing that is often helpful is to contribute PR for failing test (under `yaml/src/test/java/...../failing/`) even without fix: having easily runnable reproduction is often surprisingly helpful wrt actual fixing. Plus ensures fix actually works.

I have created PR#501 based on the successful test function SimpleGenerationTest.testLiteralBlockStyle.

**Comment by cowtowncoder** [comment]
Had a quick look at code, and style Jackson module passes to SnakeYAML is `STYLE_LITERAL` for both cases, and it is SnakeYAML that then overrides output mode.
I suspect this is because use of block style would lose information (trailing space at the last row). But at any case its SnakeYAML that decides, not Jackson module, so not sure what we could do here.

Leaving open if anyone has any ideas.

**Comment by fabians-px** [comment]
Hey @cowtowncoder and everybody else. Did you end up finding a solution for this issue or are you aware of a SnakeYAML issue regarding this topic?

**Comment by cowtowncoder** [comment]
No progress other than what was mentioned; not aware of (but also did not look for) open issues on SnakeYAML side.

**Comment by cowtowncoder** [comment]
Re-reading this, I think quoting is applied to preserve trailing white-space. So this is not a bug; will close.
